### PR TITLE
Open random document

### PR DESCRIPTION
**Pull request by robert00s**
Open random book from current folder and from unreaded documents.
Close: #3543 
test - koreader_007
test - koreader_008

### CODE REVIEW

File: frontend/apps/filemanager/filemanager.lua [hunk 1]
```diff
@@ -533,6 +543,30 @@ function FileManager:setHome(path)
     return true
 end
 
+function FileManager:openRandomFile(dir)
+    local random_file = DocumentRegistry:getRandomFile(dir, false)
+    if random_file then
+        UIManager:show(MultiConfirmBox:new {
+            text = T(_("Do you want to open %1?"), require("ffi/util").basename(random_file)),
```
Comment (Frenzie) [inline frontend/apps/filemanager/filemanager.lua]: Perhaps the name is a slight vestige, but notice that around line 30… `local util = require("ffi/util")` ;-)

File: frontend/document/documentregistry.lua [hunk 2]
```diff
@@ -26,6 +27,34 @@ function DocumentRegistry:addProvider(extension, mimetype, provider, weight)
...
 end
 
+function DocumentRegistry:getRandomFile(dir, opened, extension)
+    local DocSettings = require("docsettings")
+    if string.sub(dir, string.len(dir)) ~= "/" then
+        dir = dir .. "/"
+    end
+    local files = {}
+    local i = 0
+    math.randomseed(os.time())
```
Comment (Frenzie) [inline frontend/document/documentregistry.lua]: Wouldn't this be clearer just before where math.random is actually used? Afaik it's only if you want to give it a different seed that you'd want to wait a little to make it sufficiently distinct (in seconds for Lua's `os.time()`).

Comment (robert00s) [reply frontend/document/documentregistry.lua]: Do you mean...
```
diff --git a/frontend/document/documentregistry.lua b/frontend/document/documentregistry.lua
index 37543151..b1421786 100644
--- a/frontend/document/documentregistry.lua
+++ b/frontend/document/documentregistry.lua
@@ -34,7 +34,6 @@ function DocumentRegistry:getRandomFile(dir, opened, extension)
     end
     local files = {}
     local i = 0
-    math.randomseed(os.time())
     local ok, iter, dir_obj = pcall(lfs.dir, dir)
     if ok then
         for entry in iter, dir_obj do
@@ -51,7 +50,7 @@ function DocumentRegistry:getRandomFile(dir, opened, extension)
     else
         return nil
     end
-    logger.info(files)
+    math.randomseed(os.time())
     return dir .. files[math.random(i)]
 end 
```

Comment (Frenzie) [reply frontend/document/documentregistry.lua]: Yes, indeed. But while my asking the question implies I think that, you can disagree of course. :-)

**Pull request merged**